### DISCUSSION TITLE
Theme not being applied when used in Flatpak applications

### DISCUSSION THREAD

**Discussion by j4ger** [fcitx/fcitx5#808] [Q&A] [answered]
I'm running Fedora Kinoite which is based on an immutable core filesystem, therefore I installed fcitx5 using Flatpak. I'm able to switch to a custom theme in system apps but not in Flatpak apps (e.g. Firefox).
![image](https://github.com/fcitx/fcitx5/assets/12110829/afa1eb0b-dd21-4f63-914e-42d51c4b8147)
![image](https://github.com/fcitx/fcitx5/assets/12110829/ec75e55b-2d60-4959-b037-eb49e38ae8f7)
I've found [this issue](https://github.com/fcitx/fcitx5/issues/432) but I've set the corresponding paths to be readable in Flatseal, but nothing seems to work.
![image](https://github.com/fcitx/fcitx5/assets/12110829/70a5aedc-2edc-4211-871d-744ce85431de)
I'm very sure I missing something here...

**Comment by wengxt** [comment, marked as answer]
This is expected if you are running on wayland with GTK_IM_MODULE and QT_IM_MODULE set to fcitx.

Due to some limitation on wayland, the popup is rendered with im module itself, which requires the theme file to be readable by the application itself, which is impossible for flatpak apps.

One way to migiate this (it seems you're using GNOME), is to use this https://extensions.gnome.org/extension/261/kimpanel/ extension to change the way the UI is displayed.

**Reply by j4ger** [reply]
I see the problem now. I'm trying to install kimtoy as I'm using KDE (the Flatseal app in the screenshot above seems to be built with gtk so it might look like being in a Gnome DE). 
Unfortunately this is not that simple since there are no available RPM packages for kimtoy (nor kimpanel), I did find one published on KDE store, but it needs to be compiled and installed to /usr, which is a bit complicated to do within an immutable core filesystem. My question would be is there any prebuilt kimtoy or kimpanel (AFAIK it is also applicable to KDE, is it?) anywhere or any way to do a userspace install without touching the core filesystem? Thanks.

**Reply by wengxt** [reply]
If you use KDE and wayland, consider to make Gtk/Qt use wayland input method protocol so fcitx can render the UI on server side.

1. ensure GTK_IM_MODULE and QT_IM_MODULE are unset, otherwise some im module may still be picked up
2. systemsettings -> virtual keyboard -> select fcitx 5

**Reply by j4ger** [reply]
Thank you. The theme displays correctly after applying this method.
I tried and successfully built a rpm package of kimtoy, but it refused to pick up my fcitx5 install, is this application outdated?

**Reply by wengxt** [reply]
You need to give kimtoy ability to own some dbus service and talk to some other.

But I don’t think it will be able to display the popup in the correct position due to wayland, so probably not worth trying.

**Reply by j4ger** [reply]
I see. Once again, thanks for all the help!

**Comment by poplar-at-twilight** [comment]
Close all flatpak applications, then set empty variables for flatpak using the following commands, and then restart the application:

```
flatpak override --env=QT_IM_MODULE= --user
flatpak override --env=GTK_IM_MODULE= --user
```

The above command is only for the flatpak application of the current user. If you want to apply it to the system scope, you need sudo privileges.

Alternatively, you can write in `~/.bashrc`:

````
unset GTK_IM_MODULE
unset QT_IM_MODULE
````

**Reply by TU-O0T** [reply]
This way is really useful for me.
